rust/hyperlane-base/src/server/eigen_node.rs

```diff
@@ -0,0 +1,303 @@
+        println!("latest_checkpoint: {}", observed_checkpoint);
+        let signed_checkpoint = core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_processed", "ethereum"])
+            .get();
+
+        // logic to check if the node is healthy
+        if observed_checkpoint - signed_checkpoint <= 1 {
+            // 200 - healthy
+            StatusCode::OK
+        } else if observed_checkpoint - signed_checkpoint <= 10 {
+            // 206 - partially healthy
+            StatusCode::PARTIAL_CONTENT
+        } else {
+            // 503 - unhealthy
+            StatusCode::SERVICE_UNAVAILABLE
+        }
+    }
+
+    /// Method to return a list of services
+    /// NB: hardcoded for now
+    pub async fn node_services_handler() -> impl IntoResponse {
+        let services = vec![
+            Service {
+                id: "hyperlane-validator-indexer".to_string(),
+                name: "indexer".to_string(),
+                description: "indexes the messages from the origin chain mailbox".to_string(),
+                status: ServiceStatus::Up,
+            },
+            Service {
+                id: "hyperlane-validator-submitter".to_string(),
+                name: "submitter".to_string(),
+                description: "signs messages indexed from the indexer".to_string(),
+                status: ServiceStatus::Down,
+            },
+        ];
+        Json(services)
+    }
+
+    /// Method to return the health of a service
+    pub async fn service_health_handler(service_id: String) -> impl IntoResponse {
+        // TODO: implement logic to check if the service is healthy
+        // now just return 200
+        format!("Health check for service: {}", service_id);
+        StatusCode::OK
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use axum::http::StatusCode;
+    use prometheus::Registry;
+    use serde_json::Value;
+
+    #[tokio::test]
+    async fn test_eigen_node_api() {
+        let core_metrics =
+            Arc::new(CoreMetrics::new("dummy_relayer", 37582, Registry::new()).unwrap());
+
+        let node_api = EigenNodeAPI::new(core_metrics);
+        let app = node_api.router();
+
+        // Run the app in the background using a test server
+        let server =
+            axum::Server::bind(&"127.0.0.1:0".parse().unwrap()).serve(app.into_make_service());
+        let addr = server.local_addr();
+        let server_handle = tokio::spawn(server);
+
+        // Create a client and make a request to the `/node` endpoint
+        let client = reqwest::Client::new();
+        let res = client
+            .get(format!("http://{}/node", addr))
+            .send()
+            .await
+            .expect("Failed to send request");
+
+        // Check that the response status is OK
+        assert_eq!(res.status(), StatusCode::OK);
+
+        // check the response body if needed
+        let json: Value = res.json().await.expect("Failed to parse json");
```

Comment:
maybe was intentional to keep as a Value but another option could be to try parsing it into `NodeInfo` and comparing that

Reply:
fixed